models/news.model.js

```diff
@@ -12,7 +13,29 @@ const fetchAPI = () => {
     const filePath = path.join(__dirname, '../endpoints.json')
     return fs.readFile(filePath, "utf-8")
     .then((file) => JSON.parse(file))
-        
 }
 
-module.exports = { fetchTopics, fetchAPI }
\ No newline at end of file
+const fetchArticleByID = (article_id) => {
+    return db.query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
+    .then((fetchedArticleById) => {
+            return fetchedArticleById.rows[0]
+    })
+}
+
+const checkValidArticleId = (article_id) => {
+    if (isNaN(article_id)){
+        throw { status: 400, msg: "Bad Request" }
+    } else {
+        return db.query(`SELECT * FROM articles WHERE article_id =$1`, [article_id])
+        .then((validArticle) => {
+            if (validArticle.rows.length > 0) {
+                return true     
+            } else {
+                throw { status: 404, msg: "Article not found" }
+            }
```

Comment:
I do like this function! However, have a think about whether we could do something a little bit neater at the moment.... could we not just send a Promise.reject from our `fetchArticleByID` model, if the rows are empty, then deal with it in our error handling middleware?